tests/lib/nested.sh

```diff
@@ -302,11 +308,57 @@ create_nested_core_vm(){
 
         if [ "$BUILD_SNAPD_FROM_CURRENT" = "true" ]; then
             if is_core_16_nested_system; then
-                echo "Build from current branch is not supported yet for uc16"
+                repack_snapd_deb_into_core_snap "$WORK_DIR"
+                EXTRA_FUNDAMENTAL="$EXTRA_FUNDAMENTAL --snap $WORK_DIR/core-from-snapd-deb.snap"
+
+            elif is_core_18_nested_system; then
+                repack_snapd_deb_into_snapd_snap "$WORK_DIR"
+                EXTRA_FUNDAMENTAL="$EXTRA_FUNDAMENTAL --snap $WORK_DIR/snapd-from-deb.snap"
+
+            elif is_core_20_nested_system; then
+                snap download --basename=pc-kernel --channel="20/edge" pc-kernel
+                uc20_build_initramfs_kernel_snap "$PWD/pc-kernel.snap" "$WORK_DIR/image"
+
+                # Get the snakeoil key and cert
+                KEY_NAME=$(get_snakeoil_key)
+                SNAKEOIL_KEY="$PWD/$KEY_NAME.key"
+                SNAKEOIL_CERT="$PWD/$KEY_NAME.pem"
+
+                # Prepare the pc kernel snap
+                KERNEL_SNAP=$(ls "$WORK_DIR"/image/pc-kernel_*.snap)
+                KERNEL_UNPACKED="$WORK_DIR"/image/kernel-unpacked
+                unsquashfs -d "$KERNEL_UNPACKED" "$KERNEL_SNAP"
+                sbattach --remove "$KERNEL_UNPACKED/kernel.efi"
+                sbsign --key "$SNAKEOIL_KEY" --cert "$SNAKEOIL_CERT" "$KERNEL_UNPACKED/kernel.efi"  --output "$KERNEL_UNPACKED/kernel.efi"
+                snap pack "$KERNEL_UNPACKED" "$WORK_DIR/image"
+
+                chmod 0600 "$KERNEL_SNAP"
+                rm -f "$PWD/pc-kernel.snap"
+                rm -rf "$KERNEL_UNPACKED"
+                EXTRA_FUNDAMENTAL="--snap $KERNEL_SNAP"
+
+                # Prepare the pc gadget snap (unless provided by extra-snaps)
+                GADGET_SNAP=""
+                if [ -d extra-snaps ]; then
+                    GADGET_SNAP=$(find extra-snaps -name 'pc_*.snap')
+                fi
+                if [ -z "$GADGET_SNAP" ] && [ "$ENABLE_SECURE_BOOT" != "true" ] && [ "$ENABLE_TPM" != "true" ]; then
```

Comment:
I am not sure we want this part: [ "$ENABLE_SECURE_BOOT" != "true" ] && [ "$ENABLE_TPM" != "true" ]
In fact we had to include that to sign the gadget with the same keys used for the kernel and to start the vm. 
@anonymouse64 could you confirm that please?

Reply:
Yeah this bit could probably go away

Reply:
to be clear, what can go away is the additional checks on ENABLE_SECURE_BOOT and ENABLE_TPM, I added those because I had some additional code which I didn't push up that would modify the kernel command line for debugging only if those variables were not defined, but that was just a local debugging thing that I didn't intend to push up here

Reply:
Hm, I think I have to remove it because when it's there this nested vm test does not work:
```
$ spread -v google-nested:ubuntu-20.04-64:tests/nested/core20/
```